ui_framework/components/form/color_picker.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { ChromePicker } from 'react-color';
+
+export class KuiColorPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showColorSelector: false,
+    };
+  }
+
+  closeColorSelector = () => {
+    this.setState({ showColorSelector: false });
+  };
+
+  toggleColorSelector = () => {
+    this.setState({ showColorSelector: !this.state.showColorSelector });
+  };
+
+  handleColorSelection = (color) => {
+    this.props.onChange(color.hex);
+  };
+
+  render() {
+    const color = this.props.color || '#ffffff';
+    return (
+      <div
+        className={ this.props.className }
+        aria-label={ this.props['aria-label'] }
+        data-test-subj={ this.props['data-test-subj'] }
+      >
+        <div
+          className="kuiColorSwatch"
+          aria-label="Select a color"
+          data-test-subj="colorSwatch"
+          style={{ background: this.props.color }}
+          onClick={ this.toggleColorSelector }
+        />
+        {
+          this.state.showColorSelector ?
+            <div className="kuiColorPickerPopUp" data-test-subj="colorPickerPopup">
+              <div className="kuiColorPickerCover" onClick={ this.closeColorSelector } />
```

Comment:
Unrelated to my previous comments, I think we should look for a better way to dismiss the popup instead of by intercepting and completely blocking clicks via this popup. Ideally a user should be able to click a button or form input elsewhere on the screen, and simultaneously interact with that element and dismiss this popup. Maybe @zinckiwi could offer some suggestions for us?